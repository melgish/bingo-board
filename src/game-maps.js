export const coverAll = [
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
  [1, 1, 1, 1, 1],
]

export const bigBox = [
  [1, 1, 1, 1, 1],
  [1, 0, 0, 0, 1],
  [1, 0, 0, 0, 1],
  [1, 0, 0, 0, 1],
  [1, 1, 1, 1, 1],
]

export const smallBox = [
  [0, 0, 0, 0, 0],
  [0, 1, 1, 1, 0],
  [0, 1, 0, 1, 0],
  [0, 1, 1, 1, 0],
  [0, 0, 0, 0, 0],
]

export const bigCorners = [
  [1, 1, 0, 1, 1],
  [1, 1, 0, 1, 1],
  [0, 0, 0, 0, 0],
  [1, 1, 0, 1, 1],
  [1, 1, 0, 1, 1],
]

export const letterX = [
  [1, 0, 0, 0, 1],
  [0, 1, 0, 1, 0],
  [0, 0, 1, 0, 0],
  [0, 1, 0, 1, 0],
  [1, 0, 0, 0, 1],
]

export const letterT = [
  [1, 1, 1, 1, 1],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
]

export const cross = [
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
  [1, 1, 1, 1, 1],
  [0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0],
]

export const letterL = [
  [1, 0, 0, 0, 0],
  [1, 0, 0, 0, 0],
  [1, 0, 0, 0, 0],
  [1, 0, 0, 0, 0],
  [1, 1, 1, 1, 1],
]

export const normals = [
  [
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
    [0, 0, 0, 0, 0],
  ],
  [
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [1, 1, 1, 1, 1],
  ],
  [
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
  ],
  [
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 1, 0, 0, 0],
  ],
  [
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 1, 0, 0],
  ],
  [
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
  ],
  [
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
    [0, 0, 0, 0, 1],
  ],
  [
    [1, 0, 0, 0, 0],
    [0, 1, 0, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 0, 1],
  ],
  [
    [0, 0, 0, 0, 1],
    [0, 0, 0, 1, 0],
    [0, 0, 1, 0, 0],
    [0, 1, 0, 0, 0],
    [1, 0, 0, 0, 0],
  ],
]

export const GAME_MAPS = [
  bigBox,
  smallBox,
  bigCorners,
  letterX,
  letterT,
  cross,
  letterL,
  coverAll,
]
