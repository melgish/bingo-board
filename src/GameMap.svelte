<script>
  import { coverAll } from "./game-maps.js"
  // 5x5 array defaults to cover-all
  export let game = coverAll
</script>

<button type="button" on:click aria-label="Select Game">
  {#each game as row}
    {#each row as col}
      <div class:filled={col} />
    {/each}
  {/each}
</button>

<style>
  button {
    /* strip styling from the button */
    padding: 0;
    margin: 0;
    border: 0;
    border-radius: 0;

    display: grid;
    grid-template-rows: repeat(5 1fr);
    grid-template-columns: repeat(5, 1fr);
    grid-gap: 0.125rem;
    background-color: white;
  }
  button:hover,
  button:focus {
    outline: 2px solid red;
  }

  div {
    display: inline-block;
    width: 0.5rem;
    height: 0.5rem;
    background-color: white;
    border: 1px dashed #600;
  }
  div.filled {
    background-color: #600;
  }
</style>
