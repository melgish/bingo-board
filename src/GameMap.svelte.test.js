import { act, render } from "@testing-library/svelte"
import GameMap from "./GameMap.svelte"
import { letterT, letterX } from "./game-maps"

const BUTTON = "button"
const FILLED = "filled"

describe("GameMap", () => {
  // process the game into a string of bits
  const toBit = (div) => {
    return div.classList.contains(FILLED) ? "1" : "0"
  }
  const toBits = (dom) => {
    const divs = dom.getByRole("button").querySelectorAll("div")
    return { bits: Array.from(divs, toBit).join("") }
  }

  describe("when no map is supplied", () => {
    it("should default to coverall", () => {
      const dom = render(GameMap)

      expect(toBits(dom)).toMatchSnapshot("all-ones")
    })
  })

  describe("when map is supplied", () => {
    it("should render the map", () => {
      const dom = render(GameMap, { game: letterT })

      expect(toBits(dom)).toMatchSnapshot("letter-t")
    })
  })

  describe("when button is clicked", () => {
    let dom
    let click
    let off

    beforeEach(() => {
      dom = render(GameMap, { game: letterX })

      click = jest.fn()
      off = dom.component.$on("click", click)
    })

    afterEach(() => off())

    it("should fire an event", async () => {
      await act(() => dom.getByRole(BUTTON).click())

      expect(click).toHaveBeenCalled()
      expect(toBits(dom)).toMatchSnapshot("letter-x")
    })
  })
})
