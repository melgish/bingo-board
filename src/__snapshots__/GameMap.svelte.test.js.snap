// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameMap when button is clicked should fire an event: letter-x 1`] = `
Object {
  "bits": "1000101010001000101010001",
}
`;

exports[`GameMap when map is supplied should render the map: letter-t 1`] = `
Object {
  "bits": "1111100100001000010000100",
}
`;

exports[`GameMap when no map is supplied should default to coverall: all-ones 1`] = `
Object {
  "bits": "1111111111111111111111111",
}
`;
